src/unix/apple/macosx/app.m

```diff
@@ -542,7 +542,9 @@ static void window_mouse_button_event(Window *window, NSUInteger index, bool pre
 
 static void window_button_event(Window *window, NSEvent *event, NSUInteger index, bool pressed)
 {
-	if (window.app.pen_enabled && (event.buttonMask & NSEventButtonMaskPenTip || !index)) {
+	bool pen_in_range = event.subtype == NSTabletPointEventSubtype;
+
+	if (window.app.pen_enabled && pen_in_range && (event.buttonMask & NSEventButtonMaskPenTip || !index)) {
```

Comment:
Since `pen_in_range` is the second argument in the if struct, this always gets executed, even when `window.app.pen_enabled` is false.

```suggestion
	if (window.app.pen_enabled
		&& (event.subtype == NSTabletPointEventSubtype)
		&& (event.buttonMask & NSEventButtonMaskPenTip || !index)) {
```